### PR TITLE
feat: Adding initial gradle support

### PR DESCRIPTION
**Pull request by politeRaccoon**
Adding some gradle support.  Not removing maven (yet).  Still need to add plugin / task that will make a fatjar.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    java {
+        toolchain {
+            languageVersion.set(JavaLanguageVersion.of(17))
+        }
+        withSourcesJar()
+        withJavadocJar()
+    }
+
+    dependencies {
+        implementation("org.mitre:commons:0.0.54")
```
Comment (stellarsunset) [inline build.gradle.kts:38]: These probably get leaked into your public API right? Commons and Guava at least should be `api()` not `implementation()` then.

Comment (politeRaccoon) [reply build.gradle.kts:38]: Correct, at least on commons

File: .gitignore [hunk 2]
```diff
@@ -45,15 +56,6 @@ buildNumber.properties
 # IntelliJ
 /out/
 
-## Eclipse ###
```
Comment (Master-Move) [inline .gitignore:48]: 🪦 eclipse

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        implementation("org.apache.avro:avro:1.11.0")
+
+        implementation("com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.12.4")
+        implementation("com.fasterxml.jackson.core:jackson-core:2.15.2")
+        implementation("com.fasterxml.jackson.core:jackson-databind:2.15.2")
+        implementation("org.yaml:snakeyaml:1.26")
+
+        testImplementation("org.junit.jupiter:junit-jupiter-engine:5.10.0")
+        testImplementation("org.junit.jupiter:junit-jupiter-params:5.10.0")
+        testImplementation("org.hamcrest:hamcrest-all:1.3")
```
Comment (Master-Move) [inline build.gradle.kts:54]: hamcrest ❤️

**Pull request merged**